src/demo/layout/MainLayout.tsx

```diff
@@ -36,19 +36,22 @@ const MainLayout = (props: MainLayoutProps) => {
     collapseSidebar();
   }, []);
 
+  const handleMouseEnter = () => {
+    setCollapse(true);
+    collapseSidebar();
+  };
+
+  const handleMouseLeave = () => {
+    setCollapse(false);
+    collapseSidebar();
+  };
+
   return (
     <div className="wrapper">
       <div className="sidebar" style={{ display: "flex", height: "100%" }}>
         <Sidebar>
```

Comment:
looks like you should be using `<Sidebar collapsed={collapse}>`
`useProSidebar()` is not part of this API
https://www.npmjs.com/package/react-pro-sidebar#user-content-api

Reply:
`<Menu>` does not have API for onMouseEnter() or onMouseLeave()